components/mpas-ocean/src/shared/mpas_ocn_manufactured_solution.F

```diff
@@ -0,0 +1,247 @@
+
+      !-----------------------------------------------------------------
+      ! input/output variables
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(:,:), intent(inout) :: &
+         tend          !< Input/Output: velocity tendency 
+
+      !-----------------------------------------------------------------
+      ! output variables
+      !-----------------------------------------------------------------
+
+      integer, intent(out) :: err !< Output: error flag
+
+      !-----------------------------------------------------------------
+      ! local variables
+      !-----------------------------------------------------------------
+
+      integer :: iEdge, kmin, kmax, k
+      real (kind=RKIND) :: phase, u, v, time
+
+      ! End preamble
+      !-----------------------------------------------------------------
+      ! Begin code
+
+      if (.not. config_use_manufactured_solution) return
+
+      time = daysSinceStartOfSim*86400.0_RKIND
+
+      do iEdge = 1, nEdgesOwned
+
+         kmin = minLevelEdgeBot(iEdge)
+         kmax = maxLevelEdgeTop(iEdge)
+
+         phase = kx*xEdge(iEdge) + ky*yEdge(iEdge) - ang_freq*time
+
+         do k = kmin, kmax
+
+            u = eta0*((-fEdge(iEdge) + gravity*kx)*cos(phase) &
+                       + ang_freq*sin(phase) &
+                       - 0.5_RKIND*eta0*(kx + ky)*sin(2.0_RKIND*(phase)))
+            v = eta0*((fEdge(iEdge) + gravity*ky)*cos(phase) &
+                      + ang_freq*sin(phase) &
+                      - 0.5_RKIND*eta0*(kx + ky)*sin(2.0_RKIND*(phase))) 
+
+            tend(k,iEdge) = tend(k,iEdge) + u*cos(angleEdge(iEdge)) + v*sin(angleEdge(iEdge))
+         enddo
+
+         err = 0
+
+      enddo
+
+   !--------------------------------------------------------------------
+
+   end subroutine ocn_manufactured_solution_tend_vel!}}}
+
+!***********************************************************************
+!
+!  routine ocn_manufactured_solution_init
+!
+!> \brief  Initialize the manufactured solution tendencies 
+!> \author Steven Brus, Carolyn Begeman
+!> \date   June 2023 
+!> \details
+!>  This routine initializes constants related to the manufactured
+!>  solution tendencies
+!
+!-----------------------------------------------------------------------
+    subroutine ocn_manufactured_solution_init(domain, err)!{{{
+
+      type (domain_type), intent(inout) :: domain      
+      integer, intent(out) :: err !< Output: Error flag
+
+      type (block_type), pointer :: block
+      type (mpas_pool_type), pointer :: verticalMeshPool
+      real (kind=RKIND), dimension(:,:), pointer :: restingThickness
+
+      if (.not. config_use_manufactured_solution) return
+
+      kx = 2.0_RKIND*pii / config_manufactured_solution_wavelength_x
+      ky = 2.0_RKIND*pii / config_manufactured_solution_wavelength_y
+
+      eta0 = config_manufactured_solution_amplitude
+
```

Comment:
```suggestion

      ! This test case assumes that the restingThickness is horizontally uniform
      ! and that only one vertical level is used so only one set of indices is used here.
```
Something like this

Reply:
Thanks for adding this clarification comment!